lib/Serverless.js

```diff
@@ -97,6 +104,24 @@ class Serverless {
   }
 
   async init() {
+    if (!this._isInvokedByGlobalInstallation) {
+      if (this._shouldReportMissingServiceDeprecation) {
+        this._logDeprecation(
```

Comment:
What is the reasoning behind moving these deprecations from constructor to `init` method?

Reply:
As if this serverless instance was invoked by some older version of global installation, we may not have `isInvokedByGlobalInstallation` information available in constructor. As in first versions it was assigned to `serverless` instance after its construction, and later it was updated to be passed with `config` to constructor.

And in programmatic usage (which anyway was never advertised), in all cases `serverless = new Serverless()` should be followed with `serverless.init()`, so moving that should be ok

Reply:
Thanks for clarification :+1: